safe/reportv4/extractors/analysis_result.py

```diff
@@ -144,40 +36,130 @@ def analysis_result_extractor(impact_report, component_metadata):
     context = {}
 
     # figure out analysis report type
-    impact_layer = impact_report.impact_layer
-    """:type: qgis.core.QgsVectorLayer"""
-    analysis_layer = impact_report.analysis_layer
-    """:type: qgis.core.QgsVectorLayer"""
-    fields = impact_layer.pendingFields()
-    field_names = [field.name() for field in fields]
-
-    # TODO: Derive proper question string
-    question = None
+    hazard_layer = impact_report.hazard
+    exposure_layer = impact_report.exposure
+    impact_layer = impact_report.impact
+    analysis_layer = impact_report.analysis
 
     # find hazard class
     hazard_classification = None
-    hazard_stats = None
-    summary = None
-    evacuation_needs = None
+    summary = []
 
     analysis_feature = analysis_layer.getFeatures().next()
-    analysis_fields = analysis_layer.pendingFields()
-    if 'hazard_class' in field_names:
-        # retrieve hazard classes
-        # TODO: need a correct way to retrieve the classifications
-        hazard_classification = generic_hazard_classes
+    analysis_inasafe_fields = analysis_layer.keywords['inasafe_fields']
+    # in case there is a classification
+    if 'classification' in hazard_layer.keywords:
+
+        # retrieve hazard classification from hazard layer
+        for classification in all_hazard_classes:
+            classification_name = hazard_layer.keywords['classification']
+            if classification_name == classification['key']:
+                hazard_classification = classification
+                break
+
+        # classified hazard must have hazard count in analysis layer
         hazard_stats = []
-        for h_class in hazard_classification['classes']:
-            class_idx = analysis_layer.fieldNameIndex(h_class['key'])
-            hazard_stats.append({
-                'key': h_class['key'],
-                'name': h_class['name'],
-                'value': analysis_feature[class_idx],
-            })
+        for hazard_class in hazard_classification['classes']:
+            # hazard_count_field is a dynamic field with hazard class
+            # as parameter
+            field_key_name = hazard_count_field['key'] % (
+                hazard_class['key'], )
+            try:
+                # retrieve dynamic field name from analysis_fields keywords
```

Comment:
Thank's, this is really helpful. But in this case, I still need to do it this way, since I've seen cases that inasafe_fields doesn't include low_hazard_count, and I still need to report it.